lib/AlgodexApi.js

```diff
@@ -126,17 +126,17 @@ function _filterExistingWallets(current, addresses) {
 }
 
 const _sendTransactions = async (client, signedTransactions) => {
-  const sendAndWaitForConfirmation = async (txnBlobs) => {
-    const {txId} = await client.sendRawTransaction(txnBlobs).do();
-    await algosdk.waitForConfirmation(client, txId, 10);
-  };
-  await Promise.all(
-      signedTransactions.map((group) => {
-        logger.debug(`Sending ${group.length} Group Transactions`);
-        const rawTxns = group.map((txn) => txn.blob);
-        return sendAndWaitForConfirmation(rawTxns);
-      }),
-  );
+  const sentRawTxnIdArr = [];
+
+  for (const group of signedTransactions) {
+    logger.debug(`Sending ${group.length} Group Transactions`);
+    const rawTxns = group.map((txn) => txn.blob);
+
+    const {txId} = await client.sendRawTransaction(rawTxns).do();
+    sentRawTxnIdArr.push(txId);
+  }
+
+  await Promise.all(sentRawTxnIdArr.map((txId) => algosdk.waitForConfirmation(client, txId, 10 )));
```

Comment:
This looks good for now. In the future it can be improved by sending only the cut transactions sequentially and all the others with Promise.all since they can happen in parallel. But doesn't need to be done for the initial integration.

Reply:
Agreed! I feel like this is a happy middle ground for the sake of pushing the patch point for next3 to use and anecdotally, it feels as fast as v1 to me.